### ISSUE TITLE
Expose $DOCKER_HOST_IP

### ISSUE THREAD

**Issue by syndbg** [docker/machine#3844]
Currently if you're using `docker-machine` and already set environmental variables using `eval "$(docker-machine env <machine_name>)"`, you still have to use `docker-machine ip` to get the IP only.

You have `$DOCKER_HOST` (e.g `tcp://192.168.99.100:2376`) however it's not in an usable format to reference in `docker-compose.yml` files as an URL's host.

E.g what I want.

```
environment:
  MY_ASSET_HOST: "http://${DOCKER_HOST_IP}:3500"
```

However what I have to do is in my shell environment, set `export DOCKER_HOST_IP=$(docker-machine ip)` and only then

```
environment:
  MY_ASSET_HOST: "http://${DOCKER_HOST_IP}:3500"
```

I suggest that `docker-machine env <machine_name>` sets an ENV variable `DOCKER_HOST_IP`. 💯

**Comment by guycalledseven** [comment]
bump

**Comment by syndbg** [comment]
@guycalledseven I hope that with persistency we can convince @jeanlaurent over :)

From my point-of-view adding the env variable is no risk for Docker and no big concern. On the other hand I'm not actively developing Docker in the first place to know all the risks.

**Comment by guycalledseven** [comment]
@syndbg I agree with you. Maybe it is not needed on native *nix docker, but when using docker-machine (macos) and docker-compose with a bit complicated setup, I really need that ENV being set.

**Comment by edmorley** [comment]
See also: https://github.com/docker/machine/issues/3216#issuecomment-305586836

**Comment by DHager** [comment]
FWIW, my Windows environment (Windows 7, Docker Toolbox 1.12.2 ) possesses a `NO_PROXY` variable that might work in a pinch, but I don't know enough about its purpose to say that it's truly reliable/portable enough for everybody's use cases.

    $ env | grep 192.168
    DOCKER_HOST=tcp://192.168.99.100:2376
    NO_PROXY=192.168.99.100

Personally, I see this as a problem of:

1. I want to make it easy for other developers to work on my code
2. I want to support interactive debugging for things which occur via terminal or asynchronously
3. The debugger running in the container attempts to connect to the IDE the user has installed on their host machine
4. Whoops! No good reliable IP address for this across platforms! Now everybody's got an extra "go manually change this and remember not to check it in" step.
